prprocessor/__main__.py

```diff
@@ -373,18 +367,42 @@ async def on_pr_merge(*, pull_request: Mapping, **_kw) -> None:
     if issue_ids:
         redmine = get_redmine()
         project = redmine.project.get(config.project)
-        fixed_in_version = get_latest_open_version(project, version_prefix)
-
-        if not fixed_in_version:
-            logger.info('Unable to determine latest version for %s; prefix=%s', project.name,
-                        version_prefix)
-            return
-
-        for issue in get_issues(redmine, issue_ids):
-            if issue.project.id == project.id:
-                logger.info('Setting fixed in version for issue %s to %s', issue.id,
-                            fixed_in_version.name)
-                set_fixed_in_version(issue, fixed_in_version)
+
+        if pull_request['merged']:
+            target_branch = pull_request['base']['ref']
+            version_prefix = get_version_prefix_from_branch(target_branch)
+            if not version_prefix:
+                logger.info('Unable to set fixed in version for %s branch %s in PR %s',
+                            repository, target_branch, pull_request['number'])
+                return
+
+            if config.version_prefix:
+                version_prefix = f'{config.version_prefix}{version_prefix}'
+
+            fixed_in_version = get_latest_open_version(project, version_prefix)
+
+            if not fixed_in_version:
+                logger.info('Unable to determine latest version for %s; prefix=%s', project.name,
+                            version_prefix)
+                return
+
+            for issue in get_issues(redmine, issue_ids):
+                if issue.project.id == project.id:
+                    logger.info('Setting fixed in version for issue %s to %s', issue.id,
+                                fixed_in_version.name)
+                    set_fixed_in_version(issue, fixed_in_version)
+        else:
+            pr_url = pull_request['html_url']
+
+            for issue in get_issues(redmine, issue_ids):
+                pr_field = issue.custom_fields.get(Field.PULL_REQUEST)
+                try:
+                    new_value = pr_field.value.remove(pr_url)
+                except ValueError:
+                    logger.debug('Issue %s not linked to PR %s', issue.id, pr_url)
+                else:
+                    logger.info('Removing PR %s from issue %s', pr_url, issue.id)
+                    issue.save(custom_fields=[{'id': pr_field.id, 'value': new_value}])
```

Comment:
the old code in `close_inactive.rb` also did set the status to `NEW` and removed the assignee.

it did so unconditionally, which I think is wrong (imagine two PRs for the same issue, one gets merged, the other closed), but in the "one PR" case we can do that too?

Reply:
Tracking in https://github.com/theforeman/prprocessor/issues/184. I'd prefer to merge this now and keep it simple. It's already big enough of a change.